include/matrix3x4SIMD.h

```diff
@@ -46,6 +46,35 @@ struct matrix3x4SIMD
 		return out;
 	}
 
+	static inline matrix3x4SIMD concatenateBFollowedByAPrecisely(const matrix3x4SIMD& _a, const matrix3x4SIMD& _b)
+	{
+		__m128d r00 = _a.halfRowAsDouble(0u, 0);
+		__m128d r01 = _a.halfRowAsDouble(0u, 1);
+		__m128d r10 = _a.halfRowAsDouble(1u, 0);
+		__m128d r11 = _a.halfRowAsDouble(1u, 1);
+		__m128d r20 = _a.halfRowAsDouble(2u, 0);
+		__m128d r21 = _a.halfRowAsDouble(2u, 1);
+
+		matrix3x4SIMD out;
+
+		double tmp[4];
+
+		_mm_storeu_pd(tmp, matrix3x4SIMD::doJob_d(r00, r01, _b, 0));
+		_mm_storeu_pd(tmp+2, matrix3x4SIMD::doJob_d(r00, r01, _b, 1));
+		for (size_t i = 0u; i < 4u; ++i)
+			out.rows[0].pointer[i] = tmp[i];
```

Comment:
there's cvtpd_ps for parallel double to float conversion

add a shuffle or movelh and you can put two registers made from double conversion into 1 register